internal/services/consumer.go

```diff
@@ -96,7 +96,12 @@ func (cs *ConsumerService) Consume() {
 
 				glog.Infof("Consumer group consume starting...")
 				// this method calls the methods handler on each stage: setup, consume and cleanup
-				cs.consumerGroup.Consume(ctx, []string{cs.canaryConfig.Topic}, cgh)
+				err := cs.consumerGroup.Consume(ctx, []string{cs.canaryConfig.Topic}, cgh)
+
+				if err != nil {
+					glog.Infof("Error consuming topic: %v", err.Error())
+					return
+				}
```

Comment:
you can use the "short" one line approach here:

```go
if err := cs.consumerGroup.Consume(ctx, []string{cs.canaryConfig.Topic}, cgh); err != nil {
     glog.Infof("Error consuming topic: %v", err.Error())
     return
}
```

Other than this cosmetic change I don't think that "return" is the right way to go. It will exit the go routine and it will never called again (it's part of the Consume that happens on beginning). The error could be transient so maybe we should use a "continue" to try on the next for iteration.

Reply:
Using "continue" sounds reasonable, but maybe there should be some throttling? The looping hits Kafka very frequently otherwise.

Reply:
I reviewed the code again and as you can see, the overall `Consume()` already has a backoff mechanism based on the fact that after starting the go routing, there is a wait for the consumer to be subscribed (this happens via the `Setup` function signaling on the `ready` channel). If it doesn't happen after a timeout the delay comes into the picture and the go routine is started again. In conclusion, I think that just returning as you are proposing can work. The only minor drawback is waiting for a timeout, even if the consume returned an error but it could be useful for waiting the error, if transient, to be solved. Wdyt?

Reply:
It seems reasonable to me and it's how I have been using this in testing. As you say, the drawback is mixing the error state with a timeout state, but at least it will not repeat the attempts to consume so quickly.